Tests/Unit/Core/OxidEEEventsTest.php

```diff
@@ -0,0 +1,139 @@
+            'WDOXIDEE_APIURL',
+            'WDOXIDEE_MAID',
+            'WDOXIDEE_ISOURS',
+            'WDOXIDEE_SECRET',
+            'WDOXIDEE_THREE_D_MAID',
+            'WDOXIDEE_THREE_D_SECRET',
+            'WDOXIDEE_NON_THREE_D_MAX_LIMIT',
+            'WDOXIDEE_THREE_D_MIN_LIMIT',
+            'WDOXIDEE_LIMITS_CURRENCY',
+            'WDOXIDEE_HTTPUSER',
+            'WDOXIDEE_HTTPPASS',
+            'WDOXIDEE_BASKET',
+            'WDOXIDEE_DESCRIPTOR',
+            'WDOXIDEE_ADDITIONAL_INFO',
+        ];
+
+        foreach ($this->_aExtendedPaymentTableCols as $colName) {
+            DatabaseProvider::getDb()->execute("ALTER TABLE `oxpayments` DROP COLUMN `{$colName}`");
+        }
+
+        $this->_aExtendedOrderTableCols = [
+            'WDOXIDEE_ORDERSTATE',
+            'WDOXIDEE_FINAL',
+            'WDOXIDEE_PROVIDERTRANSACTIONID',
+            'WDOXIDEE_TRANSACTIONID',
+            'WDOXIDEE_FINALIZEORDERSTATE',
+        ];
+
+        foreach ($this->_aExtendedOrderTableCols as $colName) {
+            DatabaseProvider::getDb()->execute("ALTER TABLE `oxorder` DROP COLUMN `{$colName}`");
+        }
+
+        parent::setUp();
+    }
+
+    /**
+     * @throws \OxidEsales\Eshop\Core\Exception\DatabaseConnectionException
+     * @throws \OxidEsales\Eshop\Core\Exception\DatabaseErrorException
+     */
+    protected function tearDown()
+    {
+        DatabaseProvider::getDB()->execute("DROP TABLE IF EXISTS `" . \Wirecard\Oxid\Core\OxidEEEvents::TRANSACTION_TABLE . "`");
+        OxidEEEvents::createOrderTransactionTable();
+
+        OxidEEEvents::extendPaymentMethodTable();
+        OxidEEEvents::extendOrderTable();
+        OxidEEEvents::addPaymentMethods();
+
+        parent::tearDown();
+    }
+
+    public function testOnActivate()
+    {
+        OxidEEEvents::onActivate();
+
+        $dbMetaDataHandler = oxNew(DbMetaDataHandler::class);
+        $this->assertTrue($dbMetaDataHandler->tableExists('wdoxidee_ordertransactions'));
+
+        foreach ($this->_oXmlPaymentMethods->payment as $paymentMethod) {
+            $payment = oxNew(Payment::class);
+            $payment->load($paymentMethod->oxid);
+            $this->assertEquals(0, $payment->oxpayments__oxactive->value, $paymentMethod->oxid . " is active");
+        }
+
+        foreach ($this->_aExtendedPaymentTableCols as $colName) {
+            $this->assertTrue($dbMetaDataHandler->fieldExists($colName, 'oxpayments'));
+        }
+
+        foreach ($this->_aExtendedOrderTableCols as $colName) {
+            $this->assertTrue($dbMetaDataHandler->fieldExists($colName, 'oxorder'));
+        }
+    }
+
+    public function testOnDeactivate()
+    {
+        OxidEEEvents::onActivate();
+        OxidEEEvents::onDeactivate();
+
+        $dbMetaDataHandler = oxNew(DbMetaDataHandler::class);
+        $this->assertTrue($dbMetaDataHandler->tableExists('wdoxidee_ordertransactions'));
+
+        foreach ($this->_oXmlPaymentMethods->payment as $paymentMethod) {
```

Comment:
this assert is totally the same as the activate one. I think there is a copy/paste error. But same principle data providers is the way to go.

Reply:
sure, and that is fine. it tests that we do not delete anything, and all payment methods are deactivated.